packages/archive/src/lib/list.ts

```diff
@@ -41,38 +40,39 @@ export default async function getPostList(
 ) {
   const response = await getResponse(
     logger,
-    `https://www.luogu.com/api/discuss?page=${page}`,
+    `https://www.luogu.com/discuss?_contentOnly&page=${page}`,
     false,
   );
   const {
-    data: { result },
-  } = (await response.json()) as LegacyDiscussList;
+    currentData: {
+      posts: { result },
+    },
+  } = (await response.json()) as PostListResponse;
   const saved = Object.fromEntries(
     (
       await prisma.post.findMany({
         select: {
           id: true,
-          replies: { select: { time: true }, orderBy: { id: "desc" }, take: 1 },
+          replies: { select: { id: true }, orderBy: { id: "desc" }, take: 1 },
         },
         where: {
           id: {
             in: result
-              .filter((post) => post.SubmitTime < after)
-              .map((post) => post.PostID),
+              .filter((post) => post.time < after)
+              .map((post) => post.id),
           },
         },
       })
-    ).map(({ id, replies }) => [id, replies[0]?.time]),
+    ).map(({ id, replies }) => [id, replies[0]?.id as number | undefined]),
   );
   return result
     .filter(
       (post) =>
-        post.SubmitTime >= after ||
-        !(post.PostID in saved) ||
-        (post.LatestReply &&
-          (!saved[post.PostID] ||
-            Math.floor(saved[post.PostID].getTime() / 60000) !==
-              Math.floor(post.LatestReply.ReplyTime / 60))),
+        post.time >= after || // 晚于特定时间则一定尝试保存
+        !(post.id in saved) || // 没有保存过亦如此
+        (post.recentReply && // 有回复并
+          (!saved[post.id] || // 上一次保存时没有回复或
+            saved[post.id] !== post.recentReply.id)), // 最新回复 id 不一致
```

Comment:
```suggestion
            saved[post.id] >= post.recentReply.id)), // 没有更新的回复（该 id 随时间单调递增）
```

可能会存在删最后一层回复的情况 然后就会多轮询几次

Reply:
这里是不是应该是 `<=`？最新回复新于上次保存的回复时返回 `true`。

Reply:
嘛 刚才脑子抽了 edit 了

Reply:
他不应当取等）